src/frontend/lib/intl.ts

```diff
@@ -13,40 +13,58 @@ export interface LanguageName {
   englishName: string;
 }
 
-const translatedLocales = Object.keys(messages) as Array<TranslatedLocale>;
+export const SUPPORTED_LANGUAGES: Array<Language> = deriveSupportedLanguages(
+  Object.keys(MESSAGES) as Array<TranslatedLocale>,
+);
 
-export const supportedLanguages: LanguageName[] = translatedLocales
-  .filter(translatedLocale => {
+function deriveSupportedLanguages(
+  translatedLocales: Array<TranslatedLocale>,
+): Array<Language> {
+  const result: Array<Language> = [];
+
+  for (const locale of translatedLocales) {
     const hasAtLeastOneTranslatedString =
-      Object.keys(messages[translatedLocale]).length > 0;
-    // This will show a typescript error if the language name does not exist
-    const hasTranslatedLanguageName = languages[translatedLocale];
-    if (!hasTranslatedLanguageName) {
-      console.warn(
-        `Locale "${translatedLocale}" is not available in Mapeo because we do not have
-a language name and translations in \`src/frontend/languages.json\``,
-      );
+      Object.keys(MESSAGES[locale]).length > 0;
+
+    if (!hasAtLeastOneTranslatedString) continue;
+
+    if (!isSupportedLanguageLocale(locale)) {
+      if (process.env.APP_VARIANT === 'development') {
+        console.warn(
+          `Locale "${locale}" is not available in CoMapeo (see \`src/frontend/languages.json\`)`,
+        );
+      }
```

Comment:
figured it would be fine to update the message here a bit and also only have it logged in development builds. open to changing though